src/solvers/fv_t8code/fv.jl

```diff
@@ -0,0 +1,250 @@
+end
+
+function compute_coefficients!(u, func, t, mesh::T8codeMesh,
+                               equations, solver::FV, cache)
+    for element in eachelement(mesh, solver, cache)
+        x_node = SVector(cache.elements[element].midpoint) # Save t8code variables as SVector?
+        u_node = func(x_node, t, equations)
+        set_node_vars!(u, u_node, equations, solver, element)
+    end
+end
+
+function create_cache(mesh::T8codeMesh, equations::AbstractEquations, solver::FV, ::Any,
+                      ::Type{uEltype}) where {uEltype <: Real}
+    count_required_surfaces!(mesh)
+
+    elements = init_fv_elements(mesh, equations, solver, uEltype)
+    interfaces = init_fv_interfaces(mesh, equations, solver, elements)
+    # boundaries = init_fv_boundaries(mesh, equations, solver, elements)
+    # mortars = init_mortars(mesh, equations, basis, elements)
+
+    # fill_mesh_info!(mesh, interfaces, mortars, boundaries,
+    #                 mesh.boundary_names)
+
+    # Temporary solution array to allow exchange between MPI ranks.
+    u_tmp = init_solution!(mesh, equations)
+
+    cache = (; elements, interfaces, u_tmp)
+
+    return cache
+end
+
+function rhs!(du, u, t, mesh::T8codeMesh, equations,
+              initial_condition, boundary_conditions, source_terms::Source, solver::FV,
+              cache) where {Source}
+    (; surface_flux) = solver
+
+    # Reset du
+    @trixi_timeit timer() "reset ∂u/∂t" du.=zero(eltype(du))
+
+    # Exchange solution between MPI ranks
+    @trixi_timeit timer() "exchange_solution!" exchange_solution!(u, mesh, equations,
+                                                                  solver, cache)
+    (; elements, interfaces, u_tmp) = cache
+
+    # Prolong solution to interfaces
+    @trixi_timeit timer() "prolong2interfaces" begin
+        prolong2interfaces!(cache, u_tmp, mesh, equations, solver)
+    end
+
+    # Calculate interface fluxes
+    @trixi_timeit timer() "interface flux" begin
+        for interface in eachinterface(solver, cache)
+            element = interfaces.neighbor_ids[1, interface]
+            neighbor = interfaces.neighbor_ids[2, interface]
+            face = interfaces.faces[1, interface]
+
+            # TODO: Save normal and face_areas in interface?
+            normal = Trixi.get_variable_wrapped(elements[element].face_normals,
+                                                equations, face)
+            u_ll, u_rr = get_surface_node_vars(interfaces.u, equations, solver,
+                                               interface)
+            flux = surface_flux(u_ll, u_rr, normal, equations)
+
+            for v in eachvariable(equations)
+                flux_ = elements[element].face_areas[face] * flux[v]
+                du[v, element] -= flux_
+                if !is_ghost_cell(neighbor, mesh)
+                    du[v, neighbor] += flux_
+                end
+            end
+        end
+    end
+
+    # TODO: Boundaries
```

Comment:
I've implemented the non-periodic stuff locally. They are about 150 lines of additional code. Should I add them to this PR?